day-01/part-1/enizor.rs

```diff
@@ -0,0 +1,44 @@
+use std::env::args;
+use std::time::Instant;
+
+fn main() {
+    let now = Instant::now();
+    let output = run(&args().nth(1).expect("Please provide an input"));
+    let elapsed = now.elapsed();
+    println!("_duration:{}", elapsed.as_secs_f64() * 1000.);
+    println!("{}", output.unwrap_or_default());
+}
+
+fn run(input: &str) -> Option<usize> {
+    let mut res = 0;
+    let mut d1: usize = usize::MAX;
+    for line in input.lines() {
+        let d2: usize = line.parse().ok()?;
```

Comment:
T'as pas un intérêt à:
```suggestion
    let d2: usize = usize::MAX;
    for line in input.lines() {
        d2 = line.parse().ok()?;
```

Sinon il refait l'allocation d'une variable locale et sa destruction à chaque tour de boucle non ?

Reply:
c'est un entier, il y a rien à détruire. Normalement le compilo est suffisamment intelligent pour que ça reste le même registre